cmd/machine-controller-manager/app/controllermanager.go

```diff
@@ -240,17 +257,20 @@ func StartControllers(s *options.MCMServer,
 		nil,
 	)
 
-	controlCoreInformerFactory := coreinformers.NewFilteredSharedInformerFactory(
+	controlCoreInformerFactory := kubernetesinformers.NewFilteredSharedInformerFactory(
 		controlCoreClientBuilder.ClientOrDie("control-core-shared-informers"),
 		s.MinResyncPeriod.Duration,
 		s.Namespace,
 		nil,
 	)
 
-	targetCoreInformerFactory := coreinformers.NewSharedInformerFactory(
-		targetCoreClientBuilder.ClientOrDie("target-core-shared-informers"),
-		s.MinResyncPeriod.Duration,
-	)
+	var targetCoreInformerFactory kubernetesinformers.SharedInformerFactory
```

Comment:
instead of passing the entire SharedInformerFactory you can just initialize `NodeInformer`. We used to pass `targetCoreInformerFactory.Core().V1().Nodes()` before. Now this will either be nil in case targetCoreClientBuilder is nil or it will `targetCoreInformerFactory.Core().V1().Nodes()`

Any reason to pass the entire `SharedInformerFactory`?